src/overlays/actors/ovl_En_Mm/z_en_mm.c

```diff
@@ -34,38 +32,190 @@ const ActorInit En_Mm_InitVars = {
+    },
+    {
+        ELEMTYPE_UNK2,
+        { 0x00100000, 0x00, 0x00 },
+        { 0x01000202, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 6, 30, 0, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_8096636C[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32_DIV1000(gravity, -1200, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 270, ICHAIN_STOP),
 };
 
-#endif
-
-extern ColliderCylinderInit D_80966340;
-extern InitChainEntry D_8096636C[];
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm/EnMm_SetupAction.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm/func_80965BBC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm/EnMm_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm/EnMm_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm/func_80965D3C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm/func_80965DB4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm/func_8096611C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm/EnMm_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm/EnMm_Draw.s")
+void EnMm_SetupAction(EnMm* this, EnMmActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void func_80965BBC(EnMm* this) {
+    EnMm_SetupAction(this, func_8096611C);
+    this->actor.room = -1;
+    if (this->actor.parent->id == 0) {
+        gSaveContext.unk_1014 = 1;
+    }
+}
+
+void EnMm_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnMm* this = THIS;
+    EnMmActionFunc action;
+
+    if ((this->actor.params >= 0) && ((!(gSaveContext.weekEventReg[37] & 0x10)) ||
+                                      (gSaveContext.weekEventReg[37] & 8) || (gSaveContext.unk_1014 != 0))) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 50.0f, func_800B3FC0, 1.2f);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    if (this->actor.parent != NULL) {
+        func_80965BBC(this);
+        return;
+    }
+    if (this->actor.cutscene >= 0) {
+        action = func_80665D3C;
+    } else {
+        action = func_80965DB4;
+    }
+    EnMm_SetupAction(this, action);
+}
+
+void EnMm_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnMm* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80665D3C(EnMm* this, GlobalContext* globalCtx) {
+    s16 cutscene = ActorCutscene_GetAdditionalCutscene(this->actor.cutscene);
+
+    if (ActorCutscene_GetCanPlayNext(cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(cutscene, &this->actor);
+        EnMm_SetupAction(this, func_80965DB4);
+    } else {
+	    ActorCutscene_SetIntentToPlay(cutscene);
+    }
```

Comment:
The indentation here looks a bit funny? `ActorCutscene_StartAndSetUnkLinkFields`/`ActorCutscene_SetIntentToPlay` should have the same level of indentation. Perhaps run the formatter?